server/aesdsocket.c

```diff
@@ -142,66 +144,84 @@ void* thread_func(void *arg)
         }
       memcpy(ptr+bytes_to_write-i,buffer,i);// copying the buffer to pointer
       memset(buffer,0,buffer_size);// clearing buffer
-      if(write_flag==true)
+      if(write_flag)
+      {
+        if(strncmp(ptr, ioctl_str, 19) == 0)
+        {
+          struct aesd_seekto llseek_buffer;   
+          if(sscanf(ptr, "AESDCHAR_IOCSEEKTO:%d,%d", &llseek_buffer.write_cmd, &llseek_buffer.write_cmd_offset) !=2)
+          {
+            syslog(LOG_ERR, "Invalid argument");
```

Comment:
@malolasimman good idea to check for error but there's no need to send ioctl in this case with invalid data.